metis-dereference/metis-dereference-service/src/main/java/eu/europeana/metis/dereference/service/dao/ProcessedEntityDao.java

```diff
@@ -60,13 +60,48 @@ public ProcessedEntity get(String resourceId) {
   public void save(ProcessedEntity processedEntity) {
     try {
       final ObjectId objectId = Optional.ofNullable(processedEntity.getId())
-          .orElseGet(ObjectId::new);
+                                        .orElseGet(ObjectId::new);
       processedEntity.setId(objectId);
       retryableExternalRequestForNetworkExceptions(() -> datastore.save(processedEntity));
     } catch (DuplicateKeyException e) {
       LOGGER.info("Attempted to save duplicate record {}, race condition expected.",
           processedEntity.getResourceId());
       LOGGER.debug("Attempted to save duplicate record - exception details:", e);
+    } catch (Exception e) {
```

Comment:
It's bad practice to use catch `Exception`. We want to be as specific as possible so we know what issue we are dealing with. Beside that, catching with an `Exception` means it will catch *all* type of exceptions, which we might not want that. If you wrote `Exception` because you want to catch run time exceptions, then use `RuntimeException`, otherwise use the exception appropriate for this.